src/include/homestore/replication/repl_dev.h

```diff
@@ -101,8 +101,8 @@ class ReplDevListener {
 public:
     virtual ~ReplDevListener() = default;
 
-    void set_repl_dev(ReplDev* rdev) { m_repl_dev = rdev; }
-    virtual ReplDev* repl_dev() { return m_repl_dev; }
+    void set_repl_dev(shared< ReplDev > rdev) { m_repl_dev = rdev; }
+    virtual shared< ReplDev > repl_dev() { return m_repl_dev; }
```

Comment:
This `virtual` confuses me; is it meant to be called by a supposed child class? e.g.:
```c++
class MyRDL : public ReplDevListener {
   ...
   shared< ReplDev > repl_dev() { ...// do something; return ReplDevListener::repl_dev(); }
};
```
If this is the case I would suggest a different pattern...add a pure virtual that is used that someone _must_ implement and leave this non-virtual, or just remove the `virtual` altogether?

Reply:
this is a default action for this virtual function. it means if the derived class does not implement this function explicitly, it will use this as the default.

Reply:
Hmmm, if there is a use-case for it fine...but it adds a vtable lookup to call regardless if someone overrides it or not. Shouldn't set_repl_dev be virtual too if the getter is overrideable?